// Copyright (c) 2016 Red Hat, Inc
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
// implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

import "common.proto";

package oaktree;

message FloatingIP {
  Location location = 1;
  string id = 2;
  bool is_attached = 3;
  string fixed_ip_address = 4;
  string floating_ip_address = 5;
  string network = 6;
  string port = 7;
  enum Status {
    active = 0;
    down = 1;
    error = 2;
  }
  Status status = 8;
  map<string, string> properties = 99;
}

message FloatingIPList {
  repeated FloatingIP floating_ips = 1;
}
